src/components/forms/formr-part-b/ProfileToFormRPartBInitialValuesMapping.ts

```diff
@@ -0,0 +1,44 @@
+import { TraineeProfile } from "../../../models/TraineeProfile";
+import { FormRPartB } from "../../../models/FormRPartB";
+
+export function ProfileToFormRPartBInitialValuesMapping(
+  traineeProfile: TraineeProfile | null
+): FormRPartB | null {
+  if (!traineeProfile) {
+    return null;
+  }
+  const pd = traineeProfile.personalDetails;
+
+  const programme =
+    traineeProfile.programmeMemberships.length > 0
+      ? traineeProfile.programmeMemberships.reduce(function(a, b) {
+          return a.startDate > b.startDate ? a : b;
+        })
+      : null;
+
+  const curriculum =
+    programme && programme.curricula.length > 0
+      ? programme.curricula
+          .filter(c => c.curriculumSubType === "MEDICAL_CURRICULUM")
+          .reduce(function(a, b) {
+            return a.curriculumStartDate > b.curriculumStartDate ? a : b;
+          })
+      : null;
+
+  const model: FormRPartB = {
+    forename: pd?.forenames || "",
+    surname: pd?.surname || "",
+    gmcNumber: pd?.gmcNumber || "",
+    email: pd?.email || "",
+    localOfficeName: pd?.personOwner || "",
+    prevRevalBody: pd?.prevRevalBody || "",
+    currRevalDate: pd?.currRevalDate,
+    prevRevalDate: pd?.prevRevalDate,
+    programmeSpecialty: curriculum?.curriculumName || "",
+    dualSpecialty: curriculum?.curriculumName || "",
```

Comment:
Just had a look at the field validation, seems that the fields 'email' and 'dualSpecialty' can leave unpopulated
https://hee-tis.atlassian.net/wiki/spaces/NTCS/pages/73662481/Trainee+UI+-+Field+Validation+Form+R+Part+B

Reply:
Basically the same as Doris plus the Date of Previous Revalidation can be in the future.

Reply:
@StevenHee Not sure of the date validations atm, will have to check with Ash.